### ISSUE TITLE
IPv6 connectivity [Question]

### ISSUE THREAD

**Issue by slin31** [richardbunt/Telemachus#46]
This might be a stupid question but I'm having trouble with the Telemachus server and I think it's and issue with IPv6.  
I can use Telemachus just fine from the same PC that KSP and the server is running on and I can use it from devices in my local net like my phone. In case of the phone only as long as I open the 8085 port in iptables but that is to be expected.  
But I want to try and let others access my Telemachus data via the Internet and that's where the problems start. My ISP only issues IPv6 IPs and all IPv4 connections are tunnelled via DS-lite or whatever that is called. The point is that I apparently can't access anything (like a webserver) from the outside via IPv4, only via IPv6.  
So I tried to access Telemachus via IPv6 but it seems like it's not working with IPv6. Is that correct or is the problem elsewhere?  
I tested to access it with the IPv6 localhost equivalent (::1) to see if it's something between me and the Internet but it did not work either. I also set up an SSH connection and opened ports in the same way I did for Telemachus to see if I made a mistake there but the SSH connection works flawlessly over IPv6.  
Various online port scanners said that there was no service listening on the Telemachus port while they confirmed SSH on the SSH port.  
  
If Telemachus does work with IPv6 then I don't know what I'm doing wrong, maybe it's my ISP meddling with stuff it should leave alone after all. If it doesn't work with IPv6 then would it be possible to make it work?  
  
Anyway thank you for reading my wall of text!

**Comment by lawnmowerlatte** [comment]
Just tested and it doesn't seem to be listening on IPv6 for me either (Mac OS X 10.10.2).
    
    $ telnet ::1 8085
    Trying ::1...
    telnet: connect to address ::1: Connection refused
    telnet: Unable to connect to remote host
    $ telnet 127.0.0.1 8085
    Trying 127.0.0.1...
    Connected to localhost.
    Escape character is '^]'.

Until this is fixed, maybe you can [set up iptables or something else](https://serverfault.com/questions/276515/use-iptables-to-forward-ipv6-to-ipv4) to forward IPv6 to IPV4. 6tunnel and socat both look promising.

**Comment by slin31** [comment]
Thank you for testing this, so the problem seems to be what I suspected.  
Also thanks for the tip with forwarding 6 to 4, I didn't think of that. I will try and see if I can get that workaround working.

**Comment by slin31** [comment]
Alright I tested to forward with socat (socat TCP6-LISTEN:8085,fork,bind=[myIPv6address] TCP4:127.0.0.1:8085). The [ ] brackets around the IP are necessary AFAIK because it's IPv6 if anyone else has this problem and finds this. Also I changed the IP Telemachus uses in the config.xml from -1 to 127.0.0.1 because I had a problem where socat complained that the port (8085) was already bound or something. Changing the xml like I did should prevent Telemachus from binding to all interfaces but maybe that is not completely necessary if you use the "bind=" parameter for socat to specify your external IPv6 address.  
Anyway the workaround works for me so I guess IPv6 support is not 100% necessary but would be nice but I don't know how easy it would be to implement. Thanks for reading.
